### DISCUSSION TITLE
Salesforce Provider: OAuth vs OIDC

### DISCUSSION THREAD

**Discussion by kyleschiess** [nextauthjs/next-auth#8091] [Help]
I'm using SvelteKit here.

I was messing with the `SalesforceProvider`, trying to get the login flow to work. In my root `+hooks.server.ts` file, I added this handler:

```
const _sf_auth_handle = SvelteKitAuth({
    providers: [
        SalesforceProvider({
            clientId: import.meta.env.VITE_SFDC_CLIENT_ID,
            clientSecret: import.meta.env.VITE_SFDC_CLIENT_SECRET,
        })
    ],
    debug: import.meta.env.VITE_VERCEL_ENV !== 'production',
    secret: import.meta.env.VITE_SF_AUTH_SECRET,
    trustHost: true
})
```

Which got me this error:

```
[auth][cause]: OperationProcessingError: Unexpected ID Token returned, use processAuthorizationCodeOpenIDResponse() for OpenID Connect callback processing
```

To clear this, I opened `core/providers/salesforce.js` and changed `type` to "oidc".

Then I got this error:

```
[auth][cause]: OperationProcessingError: unexpected JWT "iss" (issuer) claim value
```

Adding the `issuer` key to my `SalesforceProvider` in `+hooks.server.ts` cleared that error. In case anyone is curious, `issuer `should be something like this: "https://YOUR_SALESFORCE_ORG_DOMAIN.my.salesforce.com".

My next error is:

```
[auth][cause]: TypeError: User id is missing in Salesforce OAuth profile response
```

But I haven't cleared it yet. Just wanted to get this info out there for anyone else who's struggling with the `SalesforceProvider`.

I have to ask, though, why is the default `type` set to "oauth" if it ends up working on "oidc"?

**Comment by kyleschiess** [comment]
About this error---

`[auth][cause]: TypeError: User id is missing in Salesforce OAuth profile response`

I noticed this is what I'd get in the profile response:

```
"profile": {
    "at_hash": "gQVVE41IMQtNzgVzsM13qw",
    "sub": "https://login.salesforce.com/id/00DHs000001St2GMAS/005Hs00000CoGalIAF",
    "aud": "3MVG9HB6vm3GZZR.Www5imFwDzK3ziLdXi5P2EVpLlpjNekbZghg7kqN4f3OjcPqqgn7xS3LQj4ToPE.KY5iR",
    "iss": "https://MY_SALESFORCE_ORG_DOMAIN.my.salesforce.com",
    "exp": 1689804727,
    "iat": 1689804607
  }
  ```

I added this key to my `SalesforceProvider`:

```
profile: (profile, token) => {
        return {  id: profile.sub, ...profile };
 }
 ```

That cleared it.

**Reply by fracarma** [reply]
This worked for me! Thanks :D I don't get why I couldn't find an open issue, it looks like a bug to me.
